beeware/briefcase
# Add initial Android support

Hi @freakboy3742 !


Would you review and/or merge?

### Out of scope (known issues)

I know about the following, and will fix in follow-ups.

- I believe that `briefcase build android` will fail to work properly on Windows. Windows will need this code to execute `gradlew.bat`. I'm going to defer that until later when I manually test on Windows.

- The Android apps crash on start. I think this is due to bugs in my cookiecutter template. For now, you can validate that an app attempts to start.

### How to test

- Create the `helloworld` app from the tutorial

- Adjust the `pyproject.toml` to have these lines:

```
[tool.briefcase.app.helloworld.android]
requires = [
    'toga-android>=0.3.0.dev18',
]
template = "https://github.com/paulproteus/cookiecutter-beeware-android"
support_package = "https://github.com/paulproteus/Python-Android-support/releases/download/v0.5/3.7.zip"
```

- Be sure to do `pip install --editable .` of this version of `briefcase` within the virtualenv for your tutorial usage

- Run:

```
briefcase create android
briefcase run android -s emulator-5554
```

- Expected behavior: The app launches, then crashes on the device after a few seconds. :)
